pkg/cmd/services.go

```diff
@@ -443,6 +443,20 @@ Run the "mobile get serviceinstances" command from this tool to see which servic
 	}
 }
 
+func findServiceInstanceByExternalName(client versioned.Interface, ns, name string) ([]v1beta1.ServiceInstance, error) {
+	sis, err := client.ServicecatalogV1beta1().ServiceInstances(ns).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	var relevantSi []v1beta1.ServiceInstance
+	for _, s := range sis.Items {
```

Comment:
Probably better extract _Filter_ function?
```go
func Filter(svcs []v1beta1.ServiceInstance, f func(v1beta1.ServiceInstance) bool) []v1beta1.ServiceInstance {
    rel := make([]v1beta1.ServiceInstance, 0)
    for _, s := range svcs {
        if f(s) {
            rel = append(rel, s)
        }
    }
    return rel
}
```
Then use it like
```go
Filter(sis.Items, func(svc v1beta1.ServiceInstance) bool {
        return svc.Spec.ClusterServiceClassExternalName == SOME_VALUE
})
```

Reply:
I think it is ok this way. Not sure we gain too much from the filter function